mitodl/open-discussions
# Recaptcha for new signups

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1065

#### What's this PR do?
Adds a recaptcha field to the signup form

#### How should this be manually tested?
- Sign up for a Google recaptcha key
- In your `.env` file, set:
 ```
  RECAPTCHA_SITE_KEY=<your_site_key>
  RECAPTCHA_SECRET_KEY=<your_secret_key>
  ``` 
- Go to the signup page.  A recpatcha form should be visible.
- If you try to submit the form without checking it, you should get a validation message.
- When you check it, it should make you pass a test such as 'Select all images with cars"
- Once it is checked, you should be able to successfully submit the form.
- Go to the login page, it should not have a recaptcha field and should work as before.
- If you remove `RECAPTCHA_SITE_KEY` from your `.env` file, the recaptcha should not appear on the signup page and the form should work as before.
- If you use an invalid `RECAPTCHA_SECRET_KEY` value, you should receive a validation error message after submitting the form ('Error validating your submission.')

<img width="509" alt="screen shot 2018-09-05 at 11 21 45 am" src="https://user-images.githubusercontent.com/187676/45110836-6614ce00-b111-11e8-996c-fa43f07d5cb8.png">

<img width="530" alt="screen shot 2018-09-05 at 1 41 07 pm" src="https://user-images.githubusercontent.com/187676/45110844-6a40eb80-b111-11e8-9736-223ad638ce1d.png">
